File: src/1-twoSum/twoSum.go
func twoSum(nums []int, target int) []int {
    index_map := make(map[int]int)
    for idx, val := range nums {
        if left_idx, ok := index_map[target - val]; ok {
            return []int{left_idx, idx}
        }
        index_map[val] = idx
    }
    return nil
}

File: src/1-twoSum/twoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        sum_map = {}
        for index, num in enumerate(nums):
            left = target - num
            if left in sum_map:
                return [sum_map[left], index]
            sum_map[num] = index
        return None